great_expectations/core/validation_definition.py

```diff
@@ -114,6 +118,21 @@ def asset(self) -> DataAsset:
     def data_source(self) -> Datasource:
         return self.asset.datasource
 
+    def is_added(self) -> tuple[bool, list[ResourceNotAddedError]]:
```

Comment:
Similar comment about refactoring the child/dependencies part into it's own function that this calls and updating the check in `run`.